### ISSUE TITLE
[Bug]: Under Windows and matplotlib 3.5.1, "key_press_event" is not handled

### ISSUE THREAD

**Issue by thomas-haslwanter** [matplotlib/matplotlib#22827]
### Bug summary

The "key_press_event" is not handled. It looks as if this event is not connected by the event handler.

- This mistake appears under Windows (Win10 & Win11), and NOT under Linux.
- It does NOT appear with Matplotlib 3.4.1, but DOES appear with Matplotlib 3.5.1

### Code for reproduction

```python
import numpy as np
import matplotlib.pyplot as plt


def on_key(event):
    print('you pressed', event.key, event.xdata, event.ydata)


if __name__ == '__main__':
    t = np.arange(0, 20, 0.1)
    x = np.sin(t)

    fig, ax = plt.subplots()

    ax.plot(t, x)
    ax.set_title('Enter a key:')
    fig.canvas.mpl_connect('key_press_event', on_key)
    plt.show()
```

### Actual outcome

Figure is plotted, but no keyboard input is handled/indicated.

### Expected outcome

Figure is plotted, AND keyboard inputs are displayed.

### Additional information

Does NOT happen in Matplotlib 3.4.1
Does NOT happen under Linux

### Operating system

Windows

### Matplotlib Version

3.5.1

### Python version

TkAgg

**Comment by tacaswell** [comment]
attn @richardsheridan 

I suspect that this is related to the event loop work?

**Comment by richardsheridan** [comment]
does #22077 fix this? It is supposed to land in 3.5.2

**Comment by richardsheridan** [comment]
To answer my own question, yes, it seems to at least for me.

**Comment by daniilS** [comment]
Yes, should be fixed in 3.5.2. @thomas-haslwanter, until then, you can use `fig.canvas.get_tk_widget().focus_set()` as a workaround.

**Comment by thomas-haslwanter** [comment]
Thank you for the quick fix!
@daniilS, your suggesstion with `focus_set()` did not work for me. I tried it at different locations, without success. But if 3.5.2. is released in the next month, that is sufficient for met.
